Vagrant

ssayed118 opened the pull request:
- Added inputs to Inspec.yml (and several control files) to make the overlays simpler to write
- Updated kitchen files to use a vagrant testing suite

ssayed118 committed:
added org

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
added inputs

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
init

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated kitchen.yml to use local ansible

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated gemfile

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated inputs + controls

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated inputs + controls

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated inspec file

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated inspec file and some controls

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
added in default conditon

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated kitchen files for testing

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 commented:
Baseline now has all the inputs, and testing each baseline category (mod, mod-hva, etc...) passed tests

ejaronne commented:
Controls SV-230336 and SV-230374 throw profile errors. Load attached jsons into Heimdall to see that now test runs, but more interestingly, the details tab doesn't completely populate for some reason.
[ssayed_rhel8_vagrant_09132022.json.txt](https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/files/9557614/ssayed_rhel8_vagrant_09132022.json.txt)

ejaronne commented:
inspec.yml:

Change:

> #SV-230244, SV-244525:
>   - name: mins_of_inactivity
>     desc: Maximum number of minutes of account inactivity before account lockout
>     type: Numeric
>     value: 15
> 
>   - name: system_inactivity_timeout # also used in SV-230353
>     desc: Maximum system inactivity timeout (time in seconds).
>     type: Numeric
>     value: 900
> 
> #SV-230244
>   - name: client_alive_interval 
>     desc: Interval of network connections associated with SSH 
>     type: Numeric
>     value: 1800

to:

#SV-230244, SV-244525:
  - name: mins_of_inactivity
    desc: Maximum number of minutes of account inactivity before account lockout
    type: Numeric
    value: 10

  - name: client_alive_interval 
    desc: Interval of network connections associated with SSH 
    type: Numeric
    value: 600

#SV-230353
  - name: system_inactivity_timeout 
    desc: Maximum system inactivity timeout (time in seconds).
    type: Numeric
    value: 900

ejaronne commented:
For SV-244525:

Use 
name: mins_of_inactivity
desc: Maximum number of minutes of account inactivity before account lockout
type: Numeric
value: 10

name: client_alive_interval
desc: Interval of network connections associated with SSH
type: Numeric
value: 600

NOT
name: system_inactivity_timeout
which is only for SV-230353

Also: there are additional places in the text not yet using these inputs.

ejaronne commented:
SV-230331:
The check text reads:
`If any temporary accounts have no expiration date set or do not expire within {""hours""=>72, ""days""=>3} hours, this is a finding.`

instead of:
`If any temporary accounts have no expiration date set or do not expire within 72 hours, this is a finding.`

similar problem for the fix text

ejaronne commented:
Affects many controls:
change
`unlock_time= #{input('unlock_time')}`
to
`unlock_time=#{input('unlock_time')}`
removing the space after the "=" sign

ssayed118 committed:
updated control files based on pr comments

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

karikarshivani commented on inspec.yml:
```diff
@@ -8,48 +8,130 @@ version: 2.1.2
+    type: Numeric
+    value: 15
+
+  - name: system_inactivity_timeout # also used in SV-230353
+    desc: Maximum system inactivity timeout (time in seconds).
+    type: Numeric
+    value: 900
```
I need to take a closer look at this input (which was previously used in SV-244525 but was replaced with `client_alive_interval` later). The only potential use of this input seems to be in SV-230353 which is already covered by `fail_interval`

ejaronne replied:
SV-230353 is a totally separate requirement, and had always used system_inactivity_timeout. I do not see SV-230353 ever using anything else.
https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/blob/fa06c0a620740e98549ffdd4089d2ef3ed7bc490/controls/SV-230353.rb#L52

SV-244525 has only ever used client_alive_interval:
https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/blob/fa06c0a620740e98549ffdd4089d2ef3ed7bc490/controls/SV-244525.rb#L79

karikarshivani replied:
We're looking at two different control files (and I might be missing context/understanding for this), but looking at the parameterized version of [SV-230353](https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230353.rb), `fail_interval` and `fail_interval_mins` are used and I'm just wondering why we are also using `system_inactivity_timeout` when `fail_interval` has the same value of 900.

If the overlays have different values for `system_inactivity_timeout` and `fail_interval`, then it's totally understandable!

karikarshivani replied:
After our discussion, we have determined to use only the `system_inactivity_timeout` input in [SV-230353](https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230353.rb) and create more inputs as necessary specifically related to `system_inactivity_timeout`

ejaronne replied:
https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/pull/1

ssayed118 committed:
fixed controls w/ profile errors

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

karikarshivani committed:
Removed redundant comment

Signed-off-by: karikarshivani <karikarshivani@gmail.com>

ejaronne commented:
inspec.yml:
known_system_accounts is defined twice. check to make sure inputs are only defined once.

ejaronne commented:
SV-230229:
replace:
`=#{input('org_name')[:acronym]},`
with
`= #{input('org_name')[:acronym]},`
because the original STIG content has a space after the = sign

ejaronne commented:
SV-230333:
replace 
    `deny =#{input('unsuccessful_attempts')}`
with
    `deny = #{input('unsuccessful_attempts')}`
because the original STIG content has a space after the = sign

ejaronne commented:
SV-230335:
replace
    fail_interval =#{input('fail_interval')}

with
    fail_interval = #{input('fail_interval')}

because the original STIG content has a space after the = sign

ejaronne commented:
SV-230363:

replace
    difok =#{input('difok')}

with
    difok = #{input('difok')}

because the original STIG content has a space after the = sign

ejaronne commented:
SV-230483:

replace
    space_left =#{input('storage_space_left')}%

with
    space_left = #{input('storage_space_left')}%

because the original STIG content has a space after the = sign

ssayed118 committed:
fixed spacing between = and inputs (per pr comments)

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated inspec.yml

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ejaronne committed:
InSpec.yml corrections

ejaronne committed:
Update SV-230353.rb

ejaronne committed:
change some inputs

set lockout_time to o for default
removed unlock_time
removed fail_interval_mins

ejaronne committed:
removed system_inactivity_timeout_minutes

ejaronne committed:
{input('system_inactivity_timeout_minutes')} to {input('system_inactivity_timeout')/60}

ejaronne committed:
change fail_interval and lockout_time input use

ejaronne committed:
test until released

ejaronne committed:
test until strings

ejaronne committed:
test then

ejaronne committed:
Update SV-230336.rb

karikarshivani committed:
Added conditional to parameterize title, check, and fix

Signed-off-by: karikarshivani <karikarshivani@gmail.com>

karikarshivani committed:
Added missing inputs

Signed-off-by: karikarshivani <karikarshivani@gmail.com>

karikarshivani committed:
Merge pull request #1 from ssayed118/ejaronne-patch-1

Ejaronne patch 1

ejaronne commented:
wrt https://github.com/CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay/pull/1#discussion_r978984485
https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230346.rb needs to be parameterized using the maxlogins_limit input.
Also: "Consider making the data type integer in the baseline (log an issue if short on time)"

ejaronne commented:
parameterize title, description, check, fix text for SV-230369, SV-230370 using input('min_len')

ejaronne committed:
eliminate redundant mins_of_inactivity

will use client_alive_interval /60 inside the control instead

ejaronne committed:
replace #{input('mins_of_inactivity')} with {input('client_alive_interval')/60}

ejaronne committed:
replace #{input('mins_of_inactivity')} with #{input('client_alive_interval')/60}

ejaronne committed:
document 230346 using maxlogins_limit

ejaronne committed:
parameterize 'ten' and '10' using #{input('maxlogins_limit')}

ejaronne commented:
SV-230359 is N/A for HVA baselines

karikarshivani commented:
> wrt [CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay#1 (comment)](https://github.com/CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay/pull/1#discussion_r978984485) https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230346.rb needs to be parameterized using the maxlogins_limit input. Also: "Consider making the data type integer in the baseline (log an issue if short on time)"

[Issue logged](https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/issues/8) for data type conversion

ssayed118 committed:
added parameters to controls

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ejaronne committed:
parameterized temp and emergency as inputs

ejaronne committed:
clarified client_alive_interval and system_inactivity_timeout

#SV-230244, SV-244525, based on SC-10
#SV-230352, SV-230353 based on AC-11

ejaronne commented:
SV-230352 should be parameterized based on the input "system_inactivity_timeout", similar to SV-230353

ejaronne commented:
- SV-230225: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_ral
- SV-230226: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_gui
- SV-230227: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_cli

ejaronne committed:
update to use emergency account inputs

ssayed118 committed:
added system_inactivity_timeout input

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
updated SV-230352

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

ssayed118 committed:
added banner inputs

Signed-off-by: Sumaa Sayed <ssayed1@umbc.edu>

karikarshivani commented on controls/SV-230335.rb:
```diff
@@ -24,23 +24,23 @@
   "
   desc  'fix', "
-    Configure the operating system to lock an account when three unsuccessful
+    Configure the operating system to lock an account when #{input('unsuccessful_attempts')} unsuccessful
 logon attempts occur in 15 minutes.
```
Parameterize 15 minutes using `#{input('fail_interval')/60}`

ejaronne replied:
fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/c1129015b32e7c89b56ac7b21ded3b310f92185f

karikarshivani commented on controls/SV-230342.rb:
```diff
@@ -37,8 +37,8 @@
     $ sudo grep pam_faillock.so /etc/pam.d/password-auth
```
**Note** paragraph in check text needs to be parameterized for `three` and `15`

ejaronne replied:
fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/082f8a8681349c96bf8cb5667ede4874a8697b28

karikarshivani commented on controls/SV-230466.rb:
```diff
@@ -12,7 +12,7 @@
 for which the system is capable of generating audit records.
 
-    DoD has defined the list of events for which RHEL 8 will provide an audit
+  #{input('org_name')[:acronym]} has defined the list of events for which RHEL 8 will provide an audit
```
This control has check text that needs to be parameterized for unsuccessful logon attempts and fail interval.

ejaronne replied:
fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/639f9115af83a3f73e368c486fe8af1d20e7a912

ejaronne committed:
Parameterize 15 minutes using #{input('fail_interval')/60}

ejaronne committed:
check text parameterized for three and 15

ejaronne committed:
parameterized for unsuccessful logon attempts and fail interval

ejaronne committed:
Update README.md

ejaronne committed:
Update README.md

karikarshivani committed:
Changed `#` to `number` to avoid accidental comment

ejaronne committed:
prep inspec.yml profile name, version, etc for final merge

ejaronne reviewed (approved):
Looks good.

ejaronne committed:
Merge branch 'main' into vagrant

ejaronne committed:
typo corrections

ejaronne committed:
typo corrections
